src/test/java/com/timgroup/statsd/UnixSocketTest.java

```diff
@@ -177,4 +220,25 @@ public void resist_dsd_timeout() throws Exception {
         assertThat(server.messagesReceived(), hasItem("my.prefix.mycount:30|g"));
         server.clear();
     }
+
+    @Test(timeout = 10000L)
+    public void testConnectionTimeout() throws InterruptedException {
+        if (transport != "unixstream") {
+            // Connection timeout is not supported for unixgram
+            return;
+        }
+
+        // Delay the `accept()` on the server
+        server.freeze();
+        client.gauge("mycount", 10);
+        // The connection timeout in the test is set to 100ms.
+        Thread.sleep(300);
+        server.unfreeze();
+
+        server.waitForMessage();
+        assertThat(server.messagesReceived(), contains("my.prefix.mycount:10|g"));
```

Comment:
What this test is testing? If it is supposed to trigger a timeout, I don't think should receive the payload.